src/web/handlers/GestureHandler.ts

```diff
@@ -329,43 +316,42 @@ export default abstract class GestureHandler implements IGestureHandler {
       return;
     }
 
-    if (this.config.needsPointerData) {
-      this.sendTouchEvent(event);
-    }
+    this.tryToSendTouchEvent(event);
   }
   protected onPointerEnter(event: AdaptedEvent): void {
-    if (this.config.needsPointerData) {
-      this.sendTouchEvent(event);
-    }
+    this.tryToSendTouchEvent(event);
   }
   protected onPointerCancel(event: AdaptedEvent): void {
-    if (this.config.needsPointerData) {
-      this.sendTouchEvent(event);
-    }
+    this.tryToSendTouchEvent(event);
 
     this.cancel();
     this.reset();
   }
   protected onPointerOutOfBounds(event: AdaptedEvent): void {
-    this.tryToSendMoveEvent(true);
-    if (this.config.needsPointerData) {
-      this.sendTouchEvent(event);
-    }
+    this.tryToSendMoveEvent(true, event);
   }
   protected onPointerMoveOver(_event: AdaptedEvent): void {
     // used only by hover gesture handler atm
   }
   protected onPointerMoveOut(_event: AdaptedEvent): void {
     // used only by hover gesture handler atm
   }
-  private tryToSendMoveEvent(out: boolean): void {
+  private tryToSendMoveEvent(out: boolean, event: AdaptedEvent): void {
     if (
       this.enabled &&
       this.active &&
       (!out || (out && !this.shouldCancelWhenOutside))
     ) {
       this.sendEvent(this.currentState, this.currentState);
     }
+
+    this.tryToSendTouchEvent(event);
```

Comment:
Shouldn't this be handled by `onPointerMove`?

Reply:
I think it is fine to leave it here. This method is called `tryToSendMoveEvent`, so sending `touchmove` seems ok. Also note that we have two cases - `onPointerMove` and `onPointerOutOfBounds`, so instead of calling `tryToSendTouchEvent` twice, we can leave it here and it will cover both cases.

Reply:
If there's something that could be changed here, I would do something like this:

```jsx

  private tryToSendMoveEvent(out: boolean, event: AdaptedEvent): void {
    if (!out || (out && !this.shouldCancelWhenOutside)) {
      if (this.enabled && this.active) {
        this.sendEvent(this.currentState, this.currentState);
      }

      this.tryToSendTouchEvent(event);
    }
  }

```

also if we simplify the first condition to make an early return we get something that is much easier to read:

```jsx

...
    if (out && this.shouldCancelWhenOutside) {
      return;
    }
...

```

This way we could send touch events only when handler config allows it.

Reply:
Actually, I think that `this.enabled` should also be moved to the check doing early return - we shouldn't be sending any events do disabled gestures.

Reply:
Yes, I agree. Done in [9f44240](https://github.com/software-mansion/react-native-gesture-handler/pull/2923/commits/9f44240383864e8296ff367d871c49ff3b475f6a)